src/routing/ic/handler.rs

```diff
@@ -97,7 +97,10 @@ pub async fn handler(
     let ic_status = IcResponseStatus::from(&resp);
 
     // Convert it into Axum response
-    let mut response = resp.canister_response.into_response();
+    let mut response = match resp.metadata.internal_error {
```

Comment:
nit: you can probably use here `.map(...).unwrap_or_else(...)` instead of a match, though not sure if it's a lot better :)